Sources/SotoCore/Credential/STSAssumeRole.swift

```diff
@@ -85,11 +85,11 @@ struct STSCredentials: AWSDecodableShape, ExpiringCredential {
 }
 
 /// Credential Provider that holds an AWSClient
-protocol CredentialProviderWithClient: CredentialProvider {
+protocol AsyncCredentialProviderWithClient: AsyncCredentialProvider {
```

Comment:
Is it worth just renaming these to be `CredentialProvider` etc without the `Async-` prefix now that everything is async by default?

Reply:
When I get to making all the CredentialProviders async I will do this.